src/tests/Microsoft.PowerFx.Core.Tests.Shared/UDFHasSameDefinitionTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Tests.Helpers;
+using Microsoft.PowerFx.Core.Texl;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests
+{
+    public class UDFHasSameDefinitionTests : PowerFxTest
+    {
+        private static readonly ReadOnlySymbolTable _primitiveTypes = ReadOnlySymbolTable.PrimitiveTypesTableInstance;
+
+        [Theory]
+        [InlineData("Foo(x: Number): Number = Abs(x);", "Foo(x: Number): Number = Abs(x);", true)]
+        
+        // test with different udf body
+        [InlineData("Foo(x: Number): Number = Abs(x);", "Foo(x: Number): Number = Abs( /*Hello*/ x);", false)]
+        [InlineData("Foo(x: Number): Number = Abs(x);", "Foo(x: Number): Number = Abs(5);", false)]
+
+        // test with different whitespace and trivials
+        [InlineData("Foo(x: Number):   Number =   Abs(x)  ;", "Foo(x:Number):Number=Abs(x);", true)]
+        [InlineData("Foo(x: Number): Number = Abs(x);", "Foo(x: /*comment*/ Number): Number = /*comment*/ Abs(x);", true)]
+
+        // test with different function names
+        [InlineData("Foo(x: Number): Number = Abs(x);", "FooBar(x: Number): Number = Abs(x);", false)]
+
+        // test with different parameter names
+        [InlineData("Foo(x: Number): Number = Abs(x);", "Foo(y: Number): Number = Abs(x);", false)]
+        [InlineData("Foo(a: Number, b: Number, c: Number): Number = a+b+c;", "Foo(a: Number, b: Number, d: Number): Number = a+b+c;", false)]
```

Comment:
>c [](http://example.com/codeflow?start=129&length=1)

`d`?

Reply:
clarified offline these are just trivial test with no body change but with param name change and to see if they are detected in the sameness check